src/components/spreadsheet/custom-columns/custom-columns-dialog.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { useCallback, useEffect } from 'react';
+import { useIntl } from 'react-intl';
+import { Box, Dialog, DialogActions, DialogContent, DialogTitle, Grid, SxProps, Theme } from '@mui/material';
+import { CancelButton, CustomFormProvider, SubmitButton, UseStateBooleanReturn } from '@gridsuite/commons-ui';
+import { useForm } from 'react-hook-form';
+import {
+    CustomColumnForm,
+    customColumnFormSchema,
+    initialCustomColumnForm,
+    TAB_CUSTOM_COLUMN,
+} from './custom-columns-form';
+
+import { yupResolver } from '@hookform/resolvers/yup';
+import CustomColumnTable from './custom-columns-table';
+import { setCustomColumDefinitions } from 'redux/actions';
+import { TABLES_NAMES } from '../utils/config-tables';
+import { useDispatch } from 'react-redux';
+import { AppDispatch } from 'redux/store';
+import { useSelector } from 'react-redux';
+import { AppState } from 'redux/reducer';
+
+export type CustomColumnDialogProps = {
+    open: UseStateBooleanReturn;
+    indexTab: number;
+};
+
+const styles = {
+    dialogContent: {
+        width: '50%',
+        height: '60%',
+        maxWidth: 'none',
+        margin: 'auto',
+    },
+    actionButtons: { display: 'flex', gap: 2, justifyContent: 'end' },
+} as const satisfies Record<string, SxProps<Theme>>;
+
+export default function CustomColumnDialog({ open, indexTab }: Readonly<CustomColumnDialogProps>) {
+    const formMethods = useForm({
+        defaultValues: initialCustomColumnForm,
+        resolver: yupResolver(customColumnFormSchema),
+    });
+
+    const { handleSubmit, reset } = formMethods;
+    const dispatch = useDispatch<AppDispatch>();
+    const columnsDefinitions = useSelector((state: AppState) => state.allCustomColumnsDefinitions);
```

Comment:
```suggestion
    const columnsDefinitions = useSelector(
        (state: AppState) => state.allCustomColumnsDefinitions[TABLES_NAMES[indexTab]]?.columns
    );
```
Also directly this ?

Reply:
Now it's retrieved from the props